lib/Elasticsearch/Tests/QueryConditionGeneratorTest.php

```diff
@@ -1045,6 +1045,125 @@ public function it_merges_has_child_queries_from_conditional_order_queries()
         self::assertMapping(['@date'], $generator->getMappings());
     }
 
+    /** @test */
+    public function it_merges_single_level_nested_queries()
+    {
+        $condition = $this->createCondition()
+            ->field('name')
+                ->addSimpleValue('Doctor')
+                ->addSimpleValue('Foo')
+            ->end()
+            ->field('restrict')
+                ->addSimpleValue('Some')
+                ->addSimpleValue('Restriction')
+            ->end()
+            ->getSearchCondition();
+
+        $generator = new QueryConditionGenerator($condition);
+        $generator->registerField('name', 'author[].name');
+        $generator->registerField('restrict', 'author[].restrict');
+
+        self::assertEquals([
+            'query' => [
+                'bool' => [
+                    'must' => [
+                        [
+                            'nested' => [
+                                'path' => 'author',
+                                'query' => [
+                                    'bool' => [
+                                        'must' => [
+                                            [
+                                                'terms' => [
+                                                    'author.name' => [
+                                                        'Doctor',
+                                                        'Foo',
+                                                    ],
+                                                ],
+                                            ],
+                                            [
+                                                'terms' => [
+                                                    'author.restrict' => [
+                                                        'Some',
+                                                        'Restriction',
+                                                    ],
+                                                ]
+                                            ],
+                                        ],
+                                    ],
+                                ],
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+        ], $generator->getQuery()->toArray());
+
+        self::assertMapping(['name', 'restrict'], $generator->getMappings());
+    }
+
+    /** @test */
+    public function it_merges_multiple_level_nested_queries()
```

Comment:
Mark as skipped, noting the GH issue which needs to be fixed for it to work.

Reply:
Done.